plotting_tools/plot_Rlxy.C

```diff
@@ -0,0 +1,253 @@
+          hist->Rebin(rebin);
+          if(weighted) {hist->Scale(rebin*int_lumi*cross_sec*BR/N_tot);}
+          else{
+            if(signal) {hist->Scale(rebin/hist->Integral());}
+            else {
+              hist->Scale(rebin/hist->Integral());
+              hist->Scale((hist->GetEntries()*int_lumi*cross_sec*BR/N_tot)/N_tot_background[full_hist_name]);
+            }
+          }
+          hist->Sumw2(false);
+
+          if(overflow) add_overflow_bin(hist, xMax);
+          
+          if(signal) {
+            hist->SetLineColor(color);
+            hist->SetLineStyle(line);
+            vector<TH1D*> hists = break_into_smaller_histograms(hist, 0.1, false);
+            for(int i=0; i<hists.size(); i++) {
+              hists[i]->SetLineColor(color);
+              hists[i]->SetLineStyle(line);
+              hists[i]->SetLineWidth(2);
+              stacks_signal[full_hist_name]->Add(hists[i]);
+            }
+          } else {
+            hist->SetLineColorAlpha(color, 0);
+            hist->SetFillColorAlpha(color, 0.7);
+            stacks_background[full_hist_name]->Add(hist);
+          }
+        
+          if(find(in_legend.begin(), in_legend.end(), file_name) == in_legend.end()){
+            if(signal){
+              if(line==kSolid) legend_sig->AddEntry(hist, title.c_str(), "l");
+            }
+            else {legend_bkg->AddEntry(hist, title.c_str(), "f");}
+            in_legend.push_back(file_name);
+          }
+          input_file->cd();
+        }
+      }
+    }
+  }
+
+  // TODO: make this possible to run with only signal or only background
+  cout << "Saving all plots in: " << output_path << endl;
+  gROOT->SetBatch(kTRUE);
+  for(auto &[hist_name, params] : hist_names){
+    for(auto prefix : prefixes){
+      for (auto category : categories) {
+        string full_hist_name = prefix + category + hist_name;
+        string bkg_hist_name = prefix + category + hist_name;
+        if(category == "os_first_") bkg_hist_name = prefix + "os_" + hist_name;
+    
+        auto [logy, logx, rebinned, rebin, xMin, xMax, yMin, yMax, xlabel, ylabel] = params;
+
+        auto canvas = new TCanvas("canvas", "canvas");
+        canvas->cd();
+        canvas->SetBottomMargin(0.15); 
+        canvas->SetLeftMargin(0.12); 
+
+        stacks_background[bkg_hist_name]->Draw();
+        stacks_signal[full_hist_name]->Draw("nostack same ][");
+
+        set_hist_layout(stacks_background[bkg_hist_name], params, weighted);
+
+        gPad->Modified();
+        gPad->RedrawAxis();
+
+        if(logy) canvas->SetLogy();
+        if(logx) canvas->SetLogx();
+
+        double x_selection = 0.68;
+        draw_legends_and_text(legend_sig, legend_bkg, selections, prefix, x_selection, weighted, false);
+        
+        canvas->Update();
+        
+        string file_name = output_path + full_hist_name + ".pdf";
+        canvas->SaveAs(file_name.c_str());
+
+        delete canvas;
+      }
+    }
+  }
```

Comment:
Ok, this is too long and therefore very difficult to read. Could you split it into smaller functions?